src/map/utils/fishingcontest.cpp

```diff
@@ -0,0 +1,716 @@
+                            "SET claimed = 1 "
+                            "WHERE name = '%s';";
+
+        int32 ret = _sql->Query(Query, PChar->getName());
+        if (ret == SQL_ERROR)
+        {
+            ShowDebug("Unable to remove award on fishing contest for player %s.", PChar->getName());
+        }
+    }
+
+    auto GetFishingContestHistory(CCharEntity* PChar) -> std::tuple<uint16, uint16, uint16, uint16>
+    {
+        if (PChar == nullptr)
+        {
+            ShowError("Attempted to retrieve Fishing Contest History on a null pointer.");
+            return std::make_tuple(0, 0, 0, 0);
+        }
+
+        const char* Query = "SELECT contest_rank_1, contest_rank_2, contest_rank_3, contest_rank_4 "
+                            "FROM char_fishing "
+                            "WHERE charid = %u;";
+        int32       ret   = _sql->Query(Query, PChar->id);
+
+        if (ret == SQL_SUCCESS && _sql->NumRows() > 0 && _sql->NextRow() == SQL_SUCCESS)
+        {
+            return std::make_tuple(_sql->GetUIntData(0), _sql->GetUIntData(1), _sql->GetUIntData(2), _sql->GetUIntData(3));
+        }
+        else
+        {
+            ShowError("Error retrieving character fishing history from database.");
+            return std::make_tuple(0, 0, 0, 0);
+        }
+    }
+
+    // Initialization
+    void InitNewContest()
+    {
+        // Select a random fish from the lua tables
+        // This is only in lua to allow for flexibility in updating fish data
+        auto selectContestFish = lua["xi"]["fishingcontest"]["selectContestFish"];
+        if (!selectContestFish.valid())
+        {
+            sol::error err = selectContestFish;
+            ShowError("fishingcontest:InitNewContest: %s", err.what());
+            return;
+        }
+        auto result = selectContestFish();
+        if (!result.valid())
+        {
+            sol::error err = result;
+            ShowError("fishingcontest:InitNewContest:(selectContestFish): %s", err.what());
+            return;
+        }
+
+        // Randomize the criteria (length/weight/both) and measurement (greatest/smallest)
+        uint8 criteria = (uint8)std::floor(xirand::GetRandomNumber(0, CONTEST_CRITERIA_BOTH + 1));
+        uint8 measure  = (uint8)std::floor(xirand::GetRandomNumber(0, CONTEST_MEASURE_SMALLEST + 1));
+
+        InitNewContest((uint16)result, (FISHING_CONTEST_CRITERIA)criteria, (FISHING_CONTEST_MEASURE)measure);
+    }
+
+    void InitNewContest(uint16 fishId, FISHING_CONTEST_CRITERIA criteria, FISHING_CONTEST_MEASURE measure)
+    {
+        // Clear any table data involving this contest
+        const char* Query = "DELETE FROM `fishing_contest`;";
+        int32       ret   = _sql->Query(Query);
+
+        if (ret == SQL_ERROR)
+        {
+            ShowDebug("Error removing contest data.");
+            return;
+        }
+
+        // Clear the fishing contest entries from cache and database
+        FishingContestEntries.clear();
```

Comment:
In the current form, this is a memory leak of all the fishing contest entries - the vector of pointers is cleared, but the memory those pointers point to is not.

You're going to refactor things to be smart pointers, so this code won't need to change - it'll work as-is. 

If you were to keep the current manual memory management this would need to look like this:
```cpp
for (auto entry : FishingContestEntries)
{
    delete entry;
}
FishingContestEntries.clear()
```